compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -852,7 +852,9 @@ export class Environment {
            */
           return (
             this.#globals.get(binding.imported) ??
-            (isHookName(binding.imported) ? this.#getCustomHookType() : null)
+            (isHookName(binding.imported) || isHookName(binding.name)
+              ? this.#getCustomHookType()
+              : null)
```

Comment:
We should probably also just define all the APIs, including unstable ones. But this is pretty reasonable. Thanks!